src/java/io/bazel/rulesscala/worker/Worker.java

```diff
@@ -0,0 +1,114 @@
+package io.bazel.rulesscala.worker;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.lang.SecurityManager;
+import java.security.Permission;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.nio.charset.StandardCharsets;
+
+import com.google.devtools.build.lib.worker.WorkerProtocol;
+
+public final class Worker {
+
+    public static interface Interface {
+	public void work(String[] args) throws Exception;
+    }
+
+    public static void workerMain(String workerArgs[], Interface workerInterface) throws Exception {
+	if (workerArgs.length > 0 && workerArgs[0].equals("--persistent_worker")) {
```

Comment:
should we extract the `then` and `else` to private methods? Right now it was a bit hard for me to find the `else` and then took a second to remember what it means

Reply:
Sure! I'm in favor of smaller methods and it's pretty easy for this to turn into a large Java mega method.

Reply:
Done!